CSharp/Library/Microsoft.Bot.Connector.Shared/StateClientEx.cs

```diff
@@ -97,9 +100,44 @@ private static DelegatingHandler[] AddJwtTokenRefresher(DelegatingHandler[] srcH
             return handlers.ToArray();
         }
 
+        private HttpClient instanceClient;
+        protected static HttpClient g_httpClient = null;
+        protected static object syncObj = new object();
+
         partial void CustomInitialize()
         {
-            ConnectorClient.AddUserAgent(this);
+            if (g_httpClient == null)
+            {
+                lock (syncObj)
+                {
+                    if (g_httpClient == null)
+                    {
+                        g_httpClient = new HttpClient();
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Microsoft-BotFramework", "3.1"));
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"(BotBuilder .Net/{GetClientVersion(this)})"));
+                        g_httpClient.DefaultRequestHeaders.ExpectContinue = false;
+                    }
+                }
+            }
+
+            // use global singleton for perf reasons

```

Comment:
It's that hack again, of swapping one in and disposing it. 

Note: You could inherit from HttpClient and create a NonDisposableHttpClient, that just ignores Dispose.